lib/matplotlib/tests/test_figure.py

```diff
@@ -1741,3 +1741,27 @@ def test_subfigure_row_order():
     sf_arr = fig.subfigures(4, 3)
     for a, b in zip(sf_arr.ravel(), fig.subfigs):
         assert a is b
+
+
+def test_subfigure_stale_propagation():
+    fig = plt.figure()
+
+    fig.draw_without_rendering()
+    assert not fig.stale
+
+    sfig1 = fig.subfigures()
+    assert fig.stale
+
+    fig.draw_without_rendering()
+    assert not fig.stale
+    assert not sfig1.stale
+
+    sfig2 = sfig1.subfigures()
+    assert fig.stale
+
+    fig.draw_without_rendering()
+    assert not fig.stale
+    assert not sfig2.stale
+
+    sfig2.stale = True
+    assert fig.stale
```

Comment:
Also the intermediate subfigure should change as well?
```suggestion
    fig.draw_without_rendering()
    assert not fig.stale
    assert not sfig1.stale
    assert not sfig2.stale

    sfig2.stale = True
    assert sfig1.stale
    assert fig.stale
```

Reply:
subfigure.figure is the root `Figure`, so the callback doesn't affect the intermediate subfigure

https://github.com/matplotlib/matplotlib/blob/b19794e126c1077b83eba90fe74ab1f67362b1ba/lib/matplotlib/figure.py#L2223

https://github.com/matplotlib/matplotlib/blob/b19794e126c1077b83eba90fe74ab1f67362b1ba/lib/matplotlib/figure.py#L65-L67

@tacaswell thinks that's OK https://github.com/matplotlib/matplotlib/issues/28384#issuecomment-2168364060

Reply:
Ah, this is more of an interim until #28177?

Reply:
Once #28177 is in we can update the callback to explicitly work on the parent figure, so then yes the interim subfigure would also be set stale.  Of course, that would have to wait for 3.10, whereas we can potentially get this fix in for 3.9.1.